.github/workflows/docs.yml

```
@@ -158,6 +176,23 @@ jobs:
           publish_dir: doc/_build/html/
           cname: docs.pyvista.org
 
+      - name: Remove non-multi version folder
+        run: rm -rf doc
+
+      - name: Deploy on main multiversion
+        if: github.event_name == 'push'
```

Let us use the same conditional for deploying:

```suggestion
        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```